### ISSUE TITLE
Crash when copying large amounts of data to clipboard

### ISSUE THREAD

**Issue by djpereira** [Microsoft/vscode#5498]
- VSCode Version: 1.0.0
- OS Version: Windows 10 v1511

Steps to Reproduce:
1. Open a large file (35MB or more).
2. Select All (Ctrl+A)
3. Copy to clipboard (Ctrl+C)

The file opens and you can navigate very quickly. That works great, but trying to copy the whole file to clipboard makes the app crash. I am not sure there are any logs that I can pull to help diagnose the problem.

![image](https://cloud.githubusercontent.com/assets/15444756/14633557/c31800da-05e3-11e6-84ef-a34fa979baea.png)

**Comment by alexandrudima** [comment]
I think it is an OOM crash. I don't think we do anything stupid on Ctrl+C:
* we create a big fat JS string with all the text
* we call event.clipboardData.setData('text/plain', text);

Most likely this crashes when creating the big fat JS string with all the text.

**Comment by alexandrudima** [comment]
I have taken the following 4MB file and copy-pasted it 12 times until I got a file that is 50MB:

[editor.main.zip](https://github.com/Microsoft/vscode/files/811435/editor.main.zip)

I then opened that file in vs code and successfully did `ctrl+a` and `ctrl+c` and had the text in the clipboard. I pasted it in a different instance of VS Code. The renderer process consumes 430MB for me in this case.

This OOM can be avoided if we would ship a 64 bit version on Windows. #21640
